src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1880,11 +1916,57 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Bind
                 case SymbolKind.Parameter:
                     {
                         var parameter = (ParameterSymbol)symbol;
-                        if (IsBadLocalOrParameterCapture(parameter, parameter.Type, parameter.RefKind))
+                        var primaryCtor = parameter.ContainingSymbol as SynthesizedPrimaryConstructor;
+
+                        if (primaryCtor is not null &&
+                            (InParameterDefaultValue ||
+                             InAttributeArgument ||
+                             this.ContainingMember() is not { Kind: not SymbolKind.NamedType, IsStatic: false } contaningMember || // We are not in an instance member
+                             (object)contaningMember.ContainingSymbol != primaryCtor.ContainingSymbol || // The member doesn't belong to our type, i.e. from nested type
+                             ((object)contaningMember != primaryCtor && contaningMember is MethodSymbol { MethodKind: MethodKind.Constructor })) && // We are in a non-primary instance constructor
+                            !IsInsideNameof)
```

Comment:
>!IsInsideNameof [](http://example.com/codeflow?start=28&length=15)

Consider moving ahead of the `(... || ... || ...)` cases for readability.

Reply:
> Consider moving ahead of the (... || ... || ...) cases for readability.

My intent is to check this last